tools/docs/example_test_helper/src/lib.rs

```diff
@@ -79,6 +80,10 @@ pub enum Error {
     ExitedCmd,
     #[error("failed to create temporary file for stdout")]
     StdoutTempfile,
+    #[error("failed to bind with given address")]
+    BindError(io::Error),
+    #[error("failed to get local socket address of the listener")]
+    AddrError(io::Error),
```

Comment:
You can use the `Io` error instead. That way we don't lose the trace of the error and we can print/log the full details.